### PR TITLE
feat: support multiple messages for EIP712

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

This pr will support multiple messages in a transaction using EIP712 sign approach.

### Rationale

By allowing multiple messages, it will provide better user experience for dApp users.

### Example

NA

### Changes

Notable changes:
* EIP712 related messages

### CODE REVIEW

**Review by Pythonberg1997 — APPROVED** [review]

File: x/auth/tx/eip712.go [hunk 1]
```diff
@@ -92,7 +92,11 @@ func GetMsgTypes(signerData signing.SignerData, tx sdk.Tx, typedChainID *big.Int
 	}
 
 	// construct the signDoc
-	msgAny, _ := codectypes.NewAnyWithValue(protoTx.GetMsgs()[0])
+	msgAnys := make([]*codectypes.Any, 0)
```
Comment (guagualvcha) [inline x/auth/tx/eip712.go:95]: ==> `msgAnys := make([]*codectypes.Any, 0, len(protoTx.GetMsgs()) )`

File: proto/cosmos/tx/v1beta1/tx.proto [hunk 2]
```diff
@@ -117,7 +116,7 @@ message SignDocEip712 {
   Fee fee = 4 [(gogoproto.nullable) = false];
 
   // msg is the msg in the EIP712 transaction.
-  google.protobuf.Any msg = 5;
+  repeated google.protobuf.Any msg = 5;
```
Comment (guagualvcha) [inline proto/cosmos/tx/v1beta1/tx.proto:119]: `msg` ==> `msgs`?

File: x/auth/tx/eip712_test.go [hunk 3]
```diff
@@ -226,3 +226,105 @@ func TestMoreMsgs(t *testing.T) {
 		require.NotNil(t, signBytes)
 	}
 }
+
+func TestMultiMsgs(t *testing.T) {
```
Comment (guagualvcha) [inline x/auth/tx/eip712_test.go:230]: image
Many testcase does not use `SignMode_SIGN_MODE_EIP_712`, can you add the line53 change?

File: x/auth/tx/eip712.go (hunk 1 above)
Comment (forcodedancing) [reply x/auth/tx/eip712.go]: fixed

File: proto/cosmos/tx/v1beta1/tx.proto (hunk 2 above)
Comment (forcodedancing) [reply proto/cosmos/tx/v1beta1/tx.proto]: renamed

File: x/auth/tx/eip712_test.go (hunk 3 above)
Comment (forcodedancing) [reply x/auth/tx/eip712_test.go:230]: added

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**